LanguageFeatures/Augmentation-libraries/augmenting_constructors_A20_t02_lib.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Redirecting factory constructors
+/// ...
+/// It is a compile-time error if:
+/// - The augmented constructor has a body.
+///
+/// @description Checks that it is a compile-time error if the augmenting
+/// constructor references itself.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_constructors_A20_t02.dart';
+
+augment class C {
+  augment factory C.foo(int x) = C.foo;
```

Comment:
So these are cycles of length 1, I think it would be nice to have a cycle of length 2. Perhaps just copy this test and introduce  `foo2`, `bar2`, ... and then let the two form a redirection cycle?

The reason why this could matter is that it can be seen locally in the declarations that introduce a cycle of length 1 that there is a problem, but for the cycles of length 2 it is necessary to visit multiple declarations in order to discover that the cycle exists.

Reply:
I believe `LanguageFeatures/Augmentation-libraries/augmenting_constructors_A20_t03.dart` does so.

Reply:
OK, thanks!